src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -165,8 +166,44 @@ public Object instanceMainPostParseArgs() {
 
         // set the temp directory as a java property, checking for existence and read/write access
         final Path p = tmpDir.toPath();
+        Path tempFilePath = null;
         try {
             p.getFileSystem().provider().checkAccess(p, AccessMode.READ, AccessMode.WRITE);
+
+            // Warn if there's anything that prevents execution in the tmp dir because some tools need that
+            // This test relies on the file system supporting posix file permissions
+            if(p.getFileSystem().supportedFileAttributeViews().contains("posix")) {
+                // Write an empty file to the tempdir
+                tempFilePath = Files.createTempFile(p, "gatk_exec_test", null);
+                // Add execute permissions
+                final Set<PosixFilePermission> executePermissions = EnumSet.of(
+                        PosixFilePermission.OWNER_EXECUTE,
+                        PosixFilePermission.GROUP_EXECUTE,
+                        PosixFilePermission.OTHERS_EXECUTE
+                );
+                final Set<PosixFilePermission> newPermissions = Files.getPosixFilePermissions(tempFilePath);
+                newPermissions.addAll(executePermissions);
+                try{
+                    Files.setPosixFilePermissions(tempFilePath, newPermissions);
+                    if(!Files.isExecutable(tempFilePath)) {
+                        logger.warn(
+                            "User has permissions to create executable files within the configured temporary directory, " +
+                                "but cannot execute those files. It is possible the directory has been mounted using the " +
+                                "'noexec' flag. This can cause issues for some GATK tools. You can specify a different " +
+                                "directory using --tmp-dir"
+                        );
+                    }
+                }
+                catch(IOException e) {
+                    logger.warn(
```

Comment:
We should probably attach the exception here.  Since we want a sane human readable warning we might want to put an additional `logger.debug(e)` beneath this instead of logging it with the user warning.